app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/vm.rb

```diff
@@ -21,4 +21,87 @@ def self.calculate_power_state(raw_power_state)
   def self.display_name(number = 1)
     n_('Instance (IBM)', 'Instances (IBM)', number)
   end
+
+  # Send a start action to IBM Cloud. Wait for state to change to started, then update the raw_power_state.
+  def raw_start
+    with_provider_connection do |vpc|
+      instance = vpc.instances.instance(ems_ref)
```

Comment:
We have a `with_provider_object` method [[ref]](https://github.com/ManageIQ/manageiq/blob/master/app/models/mixins/provider_object_mixin.rb#L10) which is intended to be used for things like this.

You can define a `def provider_obect(vpc)` that returns `vpc.instances.instance(ems_ref)`

Reply:
Like this?

```
  def provider_object(vpc)
    vpc.instances.instance(ems_ref)
  end

  # Send a start action to IBM Cloud. Wait for state to change to started, then update the raw_power_state.
  def raw_start
    with_provider_object do |instance|
      instance.actions.start
      ....
    end
  rescue
    .... logging
  end
```

Reply:
:+1: exactly